mongodbatlas/fw_provider.go

```diff
@@ -221,7 +222,7 @@ func (p *MongodbtlasProvider) Configure(ctx context.Context, req provider.Config
 	if awsRoleDefined {
 		config.AssumeRole = parseTfModel(ctx, &assumeRoles[0])
 		secret := data.SecretName.ValueString()
-		region := data.Region.ValueString()
+		region := strings.ReplaceAll(strings.ToLower(data.Region.ValueString()), "_", "-")
```

Comment:
we might want to extract this to a function so that it's more explicit what we're doing here like MongoDBRegionFormatToAWS